app/lib/search/dart_sdk_mem_index.dart

```diff
@@ -61,13 +61,19 @@ Future<SdkMemIndex?> _createDartSdkMemIndex() async {
     return await retry(
       () async {
         final index = SdkMemIndex.dart();
-        final uri = index.baseUri.resolve('index.json');
-        final rs = await http.get(uri);
-        if (rs.statusCode != 200) {
-          throw Exception('Unexpected status code for $uri: ${rs.statusCode}');
-        }
-        final content = DartdocIndex.parseJsonText(rs.body);
+        final content = DartdocIndex.parseJsonText(
```

Comment:
Should we perhaps do this as part of the build-step... and store the value in a `sdk-index.json` inside the Docker container, instead of creating it at runtime.

Similar to how we generate dart2js, sass -> css at build time. It's safer, faster and more reliable.

We could also hardcode a yaml file with dart / flutter SDK libraries/packages and descriptions. We'll have to manually update it, and we could make a test that compares to `api.dart.dev`.

I'm not sure it's wise to introduce this cross dependency, even if we supposedly fail gracefully it's probably best to be as disconnected as possible.

We could file this as an issue and solve it in a follow-up.

Reply:
I agree. There is a slight chance that the documentation is not available at the time of the deployment, so this should also have a fallback that will try to create the index after a day or so...

Will do in a follow-up...

Reply:
I don't think we need a fallback... we could also generate the documentation into a json file when we bump `runtimeVersion`.

I'm arguing that we should consider not doing this at runtime ever, because it's fragile, it's better to have a deploy fail -- than to have failure at runtime.